src/gpio.rs

```diff
@@ -1539,194 +1605,126 @@ macro_rules! pin {
     use esp_idf_sys::*;
 
     #[cfg(feature = "riscv-ulp-hal")]
     use crate::riscv_ulp_hal::sys::*;
 
+    #[cfg(not(feature = "riscv-ulp-hal"))]
+    use crate::private::notification::Notification;
+
     use crate::adc::{ADC1, ADC2};
 
     use super::*;
-    #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "alloc"))]
-    use atomic_notification::Notification;
 
     #[allow(clippy::type_complexity)]
     #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "alloc"))]
-    pub(crate) static mut ISR_HANDLERS: [Option<Box<Box<dyn FnMut()>>>; 40] = [
+    pub(crate) static mut PIN_ISR_HANDLER: [Option<Box<Box<dyn FnMut()>>>; 40] = [
         None, None, None, None, None, None, None, None, None, None, None, None, None, None, None,
         None, None, None, None, None, None, None, None, None, None, None, None, None, None, None,
         None, None, None, None, None, None, None, None, None, None,
     ];
 
     #[allow(clippy::type_complexity)]
-    #[cfg(all(not(feature = "riscv-ulp-hal"), feature = "alloc"))]
-    pub(crate) static mut PIN_NOTIFIERS: [Option<Notification>; 40] = [
-        None, None, None, None, None, None, None, None, None, None, None, None, None, None, None,
-        None, None, None, None, None, None, None, None, None, None, None, None, None, None, None,
-        None, None, None, None, None, None, None, None, None, None,
+    #[cfg(not(feature = "riscv-ulp-hal"))]
+    pub(crate) static PIN_INTER: [AtomicU8; 40] = [
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+        super::pin_inter_new(),
+    ];
+
+    #[allow(clippy::type_complexity)]
+    #[cfg(not(feature = "riscv-ulp-hal"))]
+    pub(crate) static PIN_NOTIF: [Notification; 40] = [
+        Notification::new(),
```

Comment:
I think it should be possible to simplify this initialisation by creating a `const` holding a new `Notification` and initializing the array in the shorthand way?

Reply:
Absolutely! Oh so obvious in retrospective... as in:
`pub(crate) static PIN_NOTIF: [Notification; 40] = [INIT; 40]`, where 
`const INIT: Notification = Notification::new()`

right?

I'll fix, thanks!

Reply:
Should actually be possible to apply to the other array as well, as `AtomicU8` has a const constructor as well...